### PR TITLE
HEEDLS-492 All delegates - Filter - fix

### PR DESCRIPTION
**Pull request by ibrahimmunir14**
### JIRA link
[HEEDLS-492](https://softwiretech.atlassian.net/browse/HEEDLS-492)

### Description
- Fix new filter not being added if substring of existing filters
- Fix long names overflowing and not wrapping on delegate card

### Screenshots
longname

-----
### Developer checks
I have:
- [x] Run the formatter and made sure there are no IDE errors.
- [x] Written tests for the changes (controller, data services, services, view models etc) and manually tested my work with and without JavaScript. Full manual testing guidelines can be found here: https://softwiretech.atlassian.net/wiki/spaces/HEE/pages/6703648740/Testing
- [x] Updated/added documentation in Swiki and/or Readme.
- [x] Updated my Jira ticket with information about other parts of the system that were touched as part of the MR and have to be sanity tested to ensure nothing’s broken.
- [x] Scanned over my own MR to ensure everything is as expected.

### CODE REVIEW

**Review by livzorn — APPROVED** [review]
Looks good! :slightly_smiling_face:

File: DigitalLearningSolutions.Web.Tests/Helpers/FilteringHelperTests.cs [hunk 1]
```diff
@@ -93,14 +94,17 @@ public void AddNewFilterToFilterBy_doesnt_append_with_null_new_filter()
             result.Should().Be("Test");
         }
 
-        [Test]
-        public void AddNewFilterToFilterBy_doesnt_append_with_new_filter_already_in_filterBy()
+        [TestCase("Test", "Test")]
+        [TestCase(FilterByAlphaBravoCharlie, "Group|Name|Alpha")]
+        [TestCase(FilterByAlphaBravoCharlie, "Group|Name|Bravo")]
+        [TestCase(FilterByAlphaBravoCharlie, "Group|Name|Charlie")]
+        public void AddNewFilterToFilterBy_doesnt_append_with_new_filter_already_in_filterBy(string filterBy, string newFilterValue)
```
Comment (stellake) [inline DigitalLearningSolutions.Web.Tests/Helpers/FilteringHelperTests.cs:101]: Do we need all these test cases/what different edge cases do these cover + same question about the test below?

File: DigitalLearningSolutions.Web/Scripts/searchSortFilterAndPaginate/filter.ts [hunk 2]
```diff
@@ -132,7 +132,7 @@ function getSelectedFilterFromDropdownAndResetDropdown(
 
 function addNewFilterValueToFilterBy(newFilterValue: string): void {
   const filterBy = getFilterByValue();
-  if (!filterBy.includes(newFilterValue)) {
+  if (filterBy.split(filterSeparator).indexOf(newFilterValue) === -1) {
```
Comment (stellake) [inline DigitalLearningSolutions.Web/Scripts/searchSortFilterAndPaginate/filter.ts:135]: Can we do `.split(filterSeparator).includes(newFilterValue)`?

File: DigitalLearningSolutions.Web.Tests/Helpers/FilteringHelperTests.cs (hunk 1 above)
Comment (ibrahimmunir14) [reply DigitalLearningSolutions.Web.Tests/Helpers/FilteringHelperTests.cs:101]: They are in different parts of the existing filter string (beginning, middle and end), and different substrings of an existing filter in the filter string (beginning, middle and end). Just makes sure it splits the filters and checks properly; e.g. an implementation which just uses includes() might miss a case. Not really needed since its a very simple function so I've removed some.

File: DigitalLearningSolutions.Web/Scripts/searchSortFilterAndPaginate/filter.ts (hunk 2 above)
Comment (ibrahimmunir14) [reply DigitalLearningSolutions.Web/Scripts/searchSortFilterAndPaginate/filter.ts:135]: VS was giving me an error 😠

**Review by stellake — APPROVED** [review]
Looks good! 👍
